src/dataflow-types/src/plan/threshold.proto

```diff
@@ -0,0 +1,23 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// See https://developers.google.com/protocol-buffers for what's going on here.
+
+syntax = "proto3";
+
+import "dataflow-types/src/plan.proto";
+
+package mz_dataflow_types.plan.threshold;
+
+message ProtoThresholdPlan {
+   oneof kind {
+       mz_dataflow_types.plan.ProtoArrangement basic = 1;
+       mz_dataflow_types.plan.ProtoArrangement retractions = 2;
+   }
+}
```

Comment:
I think that there should be separate `ProtoBasicThresholdPlan` and `ProtoRetractionsThresholdPlan` structs in case if those plans gain an extra field or something.

Reply:
According to our [guidelines](https://docs.google.com/document/d/1OBc3bDZa6ag3a-LoBV8JqMqJ9CKv2qQz2IVgD6TcpjI/edit#) we unroll those types

```
struct $V1($U1);

enum $T {
  Var1($V1),
}
```
becomes
```
message Proto$T {
  oneof kind {
    〚$U1〛 var_1 = 1;
  }
}
```

I think it shouldnt complicate backwards compatibility: If we add a field, then we add a nested message type with a new number and encode both in the transition phase.